From 1df5852ed6363dc12e253bbcd75595baad46126c Mon Sep 17 00:00:00 2001
From: Michael Chan <michael.chan@broadcom.com>
Date: Sun, 22 Mar 2020 16:40:04 -0400
Subject: [PATCH 3/5] bnxt_en: Free context memory after disabling PCI in probe
 error path.
Git-commit: 62bfb932a51f6d08eb409248e69f8d6428c2cabd
Patch-mainline: v5.6
References: git-fixes

Other shutdown code paths will always disable PCI first to shutdown DMA
before freeing context memory.  Do the same sequence in the error path
of probe to be safe and consistent.

Fixes: c20dc142dd7b ("bnxt_en: Disable bus master during PCI shutdown and driver unload.")
Signed-off-by: Michael Chan <michael.chan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Denis Kirjanov <denis.kirjanov@suse.com>
---
 drivers/net/ethernet/broadcom/bnxt/bnxt.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/net/ethernet/broadcom/bnxt/bnxt.c b/drivers/net/ethernet/broadcom/bnxt/bnxt.c
index f9ef93484ccc..5288ef23f103 100644
--- a/drivers/net/ethernet/broadcom/bnxt/bnxt.c
+++ b/drivers/net/ethernet/broadcom/bnxt/bnxt.c
@@ -10834,12 +10834,12 @@ static int bnxt_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 	bnxt_clear_int_mode(bp);
 
 init_err_pci_clean:
-	bnxt_free_hwrm_short_cmd_req(bp);
-	bnxt_free_hwrm_resources(bp);
-	bnxt_free_ctx_mem(bp);
-	kfree(bp->ctx);
-	bp->ctx = NULL;
-	bnxt_cleanup_pci(bp);
+        bnxt_free_hwrm_short_cmd_req(bp);
+        bnxt_free_hwrm_resources(bp);
+        bnxt_cleanup_pci(bp);
+        bnxt_free_ctx_mem(bp);
+        kfree(bp->ctx);
+        bp->ctx = NULL;
 
 init_err_free:
 	free_netdev(dev);
-- 
2.16.4

